test/eni/IPRelease.go

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package eniTest
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/check.v1"
+
+	"github.com/cilium/cilium/pkg/logging"
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+	"github.com/cilium/cilium/test/nodegroups"
+)
+
+func getAllocatedIPsOnNode(kubectl *helpers.Kubectl, nodeName string) (int, error) {
+	cn := kubectl.GetCiliumNode(nodeName)
+	if cn == nil {
+		return 0, fmt.Errorf("Unable to fetch cn %s", nodeName)
+	}
+	return len(cn.Spec.IPAM.Pool), nil
+}
+
+func hasRequiredIPsAllocated(kubectl *helpers.Kubectl, nodeName string, n int) func() bool {
+	return func() bool {
+		cn := kubectl.GetCiliumNode(nodeName)
+		if cn == nil {
+			return false
+		}
+		return len(cn.Spec.IPAM.Pool) == n
+	}
+}
+
+func hasRequiredIPBuffer(kubectl *helpers.Kubectl, nodeName string, n int) func() bool {
+	return func() bool {
+		cn := kubectl.GetCiliumNode(nodeName)
+		if cn == nil {
+			return false
+		}
+		return (len(cn.Spec.IPAM.Pool) - len(cn.Status.IPAM.Used)) == n
+	}
+}
+
+func hasNoIPsInHandshake(kubectl *helpers.Kubectl, nodeName string) func() bool {
+	return func() bool {
+		cn := kubectl.GetCiliumNode(nodeName)
+		if cn == nil {
+			return false
+		}
+		return len(cn.Status.IPAM.ReleaseIPs) == 0
+	}
+}
+
+var _ = Describe("EniIPReleaseHandshakeTest", func() {
+	var (
+		kubectl           *helpers.Kubectl
+		ciliumCli         *helpers.CiliumCli
+		deploymentManager *helpers.DeploymentManager
+		releaseTestNs     = "cilium-integration-tests"
+	)
+
+	BeforeAll(func() {
+		var log = logging.DefaultLogger
+		var logger = logrus.NewEntry(log)
+		kubectl = helpers.CreateKubectl("", logger)
+		ciliumCli = helpers.CreateCiliumCli(logger)
+		deploymentManager = helpers.NewDeploymentManager()
+		deploymentManager.SetKubectl(kubectl)
+	})
+
+	AfterAll(func() {
+		deploymentManager.DeleteAll()
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport("cilium status", "cilium endpoint list")
+	})
+
+	JustAfterEach(func() {
+		kubectl.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
+	})
+
+	It("Test IP release abort", func() {
+		By("Enabling aws excess IP release feature and restarting cilium operator")
+		err := ciliumCli.UpdateCiliumConfig("aws-release-excess-ips", "true", false)
+		Expect(err).NotTo(HaveOccurred())
+		err = ciliumCli.UpdateCiliumConfig("excess-ip-release-delay", "30", false)
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
I'd argue that this probably belongs in the `Before()` block of the test, rather than as part of the "test" which is what I think of as the `It()` block. Ideally, it should be nice and short with mostly assertions.